### DISCUSSION TITLE
Embedding a Font

### DISCUSSION THREAD

**Discussion by SwyrWolf** [pdfcpu/pdfcpu#681] [Q&A]
I'm trying to use PDFCPU to autofill out a form with JSON data.

I'm using Source Sans 3, I got the CLI to install the font and remove the error of "can't find font" but on export it uses the right text, just not any font.

How do I get it to embed the font it uses to fill out forms?

**Comment by hhrutter** [comment]
You can't.
Imposing a different font during filling is not supported as of now.
Usually a form field (taking precedence over) and the form dict define the font to use to render a form field value.

I don't know your usecase,  but you can always create the form yourself using JSON and the pdfcpu create command.

**Reply by SwyrWolf** [reply]
> You can't.
> 
> Imposing a different font during filling is not supported as of now.
> 
> Usually a form field (taking precedence over) and the form dict define the font to use to render a form field value.
> 
> 
> 
> I don't know your usecase,  but you can always create the form yourself using JSON and the pdfcpu create command.
> 

The initial form is created in InDesign and has everything setup as an interactive PDF. When using acrobat or bluebeam to fill out the form, the font designated (Source Sans Pro) for the text fields is embedded. 

When I first attempted to use pdfcpu to fill out the form it gave me an error because source sans pro was not installed. So I used the cli program to install the font resulting in that error to go away. However, it still didn't fill in the fields with the correct font, instead it would be random symbols or the software would use a default font.

The use case I was trying to make was a cover page for submittals where none of the design changes just the values in those fields. For example the transmittal number or date.
It's essentially a template.

While I couldn't figure out how to get this to work with pdfcpu, I managed to do it with another package called "fillform" by "desertbit" which requires "PDFtk Server".
